src/core/page_usage_stats.h

```diff
@@ -93,6 +101,8 @@ class PageUsage {
 
   // For use in testing, forces reallocate check to always return true
   bool force_reallocate_{false};
+
+  TypeAndDelta last_delta_;
```

Comment:
It may be cleaner to change the return type for `DefragIfNeeded` so that it returns the object type, the delta and the fact whether defrag was conducted instead of carrying this state around, but then the signature will need to be changed in a larger set of methods, most of which do not change in size on defragmentation.

Reply:
this is no longer needed as object's `bytes_used` field is used to calculate the change